samcli/lib/build/build_strategy.py

```diff
@@ -166,6 +168,7 @@ def build_single_function_definition(self, build_definition: FunctionBuildDefini
         result = self._build_function(
             build_definition.get_function_name(),
             build_definition.codeuri,  # type: ignore
+            build_definition.imageuri,
```

Comment:
Nit: Similar to the other optional parameters, should this also have a `# type: ignore?`

Reply:
They appear to have that annotation because they're _secretly_ optional. The parameter is typed as `str`, but the argument is passed as `Optional[str]`, and this quiets the linter. To be consistent, I'd type `imageuri` the same – but since it _is_ optional, being truthful about it seemed better.